### PR TITLE
Make sure that Rollup's `dynamicRequireTargets` are included

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This is a follow-up to https://github.com/babel/babel/pull/12819: that PR was going in the good direction, but it didn't actually fix the issue: Rollup wasn't injecting the `dynamicRequireTargets` anywhere, and we didn't notice it because apparently it fallbacks to Node's `require()` function (which works on CI). [You can see it in #12819's repl](https://babeljs.io/repl/build/41157/#?browsers=%3E0.5%25%2C%20not%20ie%2010&build=&builtIns=false&spec=false&loose=false&code_lz=G4QwTgBAHhC8EHoA6AHA3gGQL4IK4G4g&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env&prettier=false&targets=&version=7.12.18%2Bpr.12819&externalPlugins=).

I noticed it when rebasing https://github.com/babel/babel/pull/12015: that PR needed another `dynamicRequireTarget`, but it caught the error because it was for a relative path which made Node's resolution algorithm fail.

I particularly dislike this solution, but it's the only thing I could came up with to make it work.

**I'd like to include this in the next version, since it's a follow-up to #12819 which has already been merged.**

Repl for this PR: https://babeljs.io/repl/build/41352/#?browsers=%3E0.5%25%2C%20not%20ie%2010&build=&builtIns=false&spec=false&loose=false&code_lz=G4QwTgBAHhC8EHoA6AHA3gGQL4IK4G4g&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env&prettier=false&targets=&version=7.12.18%2Bpr.12839&externalPlugins=

### CODE REVIEW

File: Gulpfile.mjs [hunk 1]
```diff
@@ -312,7 +314,7 @@ function buildRollup(packages, targetBrowsers) {
                 "./packages/babel-helper-create-regexp-features-plugin",
                 "regexpu-core",
                 "regenerate-unicode-properties"
-              ) + "/../**",
+              ) + "/**/*.js",
```
Comment (nicolo-ribaudo) [inline Gulpfile.mjs:317]: Without the explicit `.js` extension it was trying to parse the LICENSE file as JS.

File: packages/babel-standalone/src/index.js [hunk 2]
```diff
@@ -10,6 +10,8 @@
 /* global VERSION */
 /* eslint-disable max-len */
 
+import "./dynamic-require-entrypoint.cjs";
```
Comment (nicolo-ribaudo) [inline packages/babel-standalone/src/index.js:13]:
```suggestion
// $FlowIgnore
import "./dynamic-require-entrypoint.cjs";
```

File: .yarn/patches/@rollup__plugin-commonjs.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+diff --git a/dist/index.js b/dist/index.js
+index 712f6a7d81b115d468a764b4139caa00d6cfc358..73fbf004217f3d44b6420d3082a0846b53e00f4c 100644
+--- a/dist/index.js
++++ b/dist/index.js
+@@ -1626,7 +1626,7 @@ function commonjs(options = {}) {
+   const sourceMap = options.sourceMap !== false;
+
+   function transformAndCheckExports(code, id) {
+-    if (isDynamicRequireModulesEnabled && this.getModuleInfo(id).isEntry) {
++    if (isDynamicRequireModulesEnabled && id.endsWith("/dynamic-require-entrypoint.cjs")) {
```
Comment (JLHwung) [inline .yarn/patches/@rollup__plugin-commonjs.patch:12]: I am not familiar with the rollup plugin commonjs codebase, but I guess it doesn't hurt to:
```js
this.getModuleInfo(id).isEntry || id.endsWith("/dynamic-require-entrypoint.cjs")
```
just ensure we are not replacing this bug with another bug. 😄

**Review by JLHwung — APPROVED** [review]
Black magic.

File: .yarn/patches/@rollup__plugin-commonjs.patch (hunk 3 above)
Comment (nicolo-ribaudo) [reply .yarn/patches/@rollup__plugin-commonjs.patch:12]:
```suggestion
+    if (isDynamicRequireModulesEnabled && (this.getModuleInfo(id).isEntry || id.endsWith("/dynamic-require-entrypoint.cjs"))) {
```

File: packages/babel-standalone/src/dynamic-require-entrypoint.cjs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/*
+We bundle @babel/standalone using the dynamicRequireTargets option.
+This option needs to inject requrie() calls to all the modules that
```
Comment (existentialism) [inline packages/babel-standalone/src/dynamic-require-entrypoint.cjs:3]:
```suggestion
This option needs to inject require() calls to all the modules that
```

Comment (existentialism) [reply packages/babel-standalone/src/dynamic-require-entrypoint.cjs:3]: super small nit

**Review by existentialism — APPROVED** [review]
🪄

File: packages/babel-standalone/src/dynamic-require-entrypoint.cjs (hunk 4 above)
Comment (nicolo-ribaudo) [reply packages/babel-standalone/src/dynamic-require-entrypoint.cjs:3]: Whops thanks

**Pull request merged**